### DISCUSSION TITLE
Multiple VM like container instances with IPC

### DISCUSSION THREAD

**Discussion by al-sabr** [CppMicroServices/CppMicroServices#636] [Ideas]
This is a continuation from this thread : https://github.com/CppMicroServices/CppMicroServices/issues/633

I would like to know if it is possible to run 2 instance of CppMicroServices and let them communicate with each others with some kind of IPC?

**Comment by jeffdiclemente** [comment]
Yes, it is possible to do this. [OSGI Remote Services](https://docs.osgi.org/specification/osgi.cmpn/7.0.0/service.remoteservices.html) is a specification which can be followed to accomplish two or more CppMicroServices instances communicating with each other.

As of right now, the CppMicroServices framework does not provide an out of the box way for 2 or more instances of CppMicroServices to communicate with one another. CppMicroServices users would need to develop their own solution. We do welcome any contributions to the CppMicroServices framework. See our [RFC process](https://github.com/CppMicroServices/rfcs/blob/master/README.md) and [contributing guidelines](https://github.com/CppMicroServices/CppMicroServices/blob/development/CONTRIBUTING.rst) if you would like to contribute. This is of interest to us however currently there are other priorities and not enough developers to do the work.
